File: ResourceManager/Models/Configurations/RepositoryTagConfiguration.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System.Collections;
using System.Collections.Generic;
using ResourceManager.Configuration;

namespace ResourceManager.Models.Configurations
{
    public class RepositoryTagConfiguration : IEntityTypeConfiguration<RepositoryTag>
    {
        public void Configure(EntityTypeBuilder<RepositoryTag> builder)
        {
            builder.ToTable("RepositoryTags");
            builder.HasKey(x => new { x.RepositoryID, x.TagID });

            builder.HasOne(x => x.Repository)
                .WithMany(x => x.RepositoryTags)
                .HasForeignKey(x => x.RepositoryID);

            builder.HasOne(x => x.Tag)
                .WithMany(x => x.RepositoryTags)
                .HasForeignKey(x => x.TagID);

        }
    }
}

File: ResourceManager/Models/Configurations/RepositoryTagConfiguration.vb
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
Imports Microsoft.EntityFrameworkCore

Namespace ResourceManager.Models.Configurations
    Public Class RepositoryTagConfiguration
        Implements IEntityTypeConfiguration(Of RepositoryTag)
        Public Sub Configure(ByVal builder As EntityTypeBuilder(Of RepositoryTag)) Implements IEntityTypeConfiguration(Of RepositoryTag).Configure
            builder.ToTable("RepositoryTags")
            builder.HasKey(Function(x) New With {x.RepositoryID, x.TagID
            })

            builder.HasOne(Function(x) x.Repository).WithMany(Function(x) x.RepositoryTags).HasForeignKey(Function(x) x.RepositoryID)

            builder.HasOne(Function(x) x.Tag).WithMany(Function(x) x.RepositoryTags).HasForeignKey(Function(x) x.TagID)

        End Sub
    End Class
End Namespace
